File: V5.0/ZDK2/src/nl/zeesoft/zdk/dai/predict/PredictorConfig.java
package nl.zeesoft.zdk.dai.predict;

import java.util.ArrayList;
import java.util.List;

import nl.zeesoft.zdk.dai.ObjMapComparator;
import nl.zeesoft.zdk.dai.cache.CacheIndexesGenerator;

public class PredictorConfig {
	public ObjMapComparator		comparator			= new ObjMapComparator();
	public int					maxHistorySize		= 1000;
	public int					maxMsLoggerSize		= 1000;

	public List<Integer>		cacheIndexes		= CacheIndexesGenerator.generate();
	public List<CacheConfig>	cacheConfigs		= new ArrayList<CacheConfig>();
	
	public PredictorConfig() {
		float subtract = 0F;
		int maxSize = 10000;
		for (int i = 0; i < 3; i++) {
			CacheConfig cc = new CacheConfig();
			cc.mergeSimilarity = cc.mergeSimilarity - subtract;
			cc.maxSize = maxSize;
			if (cc.mergeSimilarity < 1F) {
				cc.linked = true;
				cc.linkedMergeSimilarity = (1F + cc.mergeSimilarity) / 2F;
				cc.linkedMaxSize = cc.maxSize / 2;
			}
			cacheConfigs.add(cc);
			subtract += 0.1F;
			maxSize = (int)((float)maxSize * (subtract * 2));
		}
	}
}

File: NodeJS/predictive-memory/src/predict/PredictorConfig.js
const Comparator = require('../Comparator');
const IndexGenerator = require('./IndexGenerator');
const CacheConfig = require('../cache/CacheConfig');

function PredictorConfig(size, depth) {
  this.maxHistorySize = size || 129;
  this.comparator = new Comparator();

  this.cacheIndexes = (new IndexGenerator(depth)).generate();
  this.cacheConfig = new CacheConfig();
  this.cacheConfig.initiatlizeDefault();
  this.cacheQueryOptions = this.cacheConfig.getQueryOptions();

  this.maxPredictionHistorySize = size || 129;

  this.setComparator = (com) => {
    this.comparator = com;
    this.cacheConfig.setComparator(com);
  };

  this.setComparator(this.comparator);
}
module.exports = PredictorConfig;
